test/setup/versions.tf

```diff
@@ -19,11 +19,11 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = "~> 3.53.0"
+      version = "~> 4.0"
```

Comment:
@morgante Thought might help this, but do we even use test/setup at the kitchen tests?

Reply:
We do use test/setup in CI. It's the stage that is failing:
```
Initializing modules...
Downloading terraform-google-modules/project-factory/google 10.3.2 for project...
- project in .terraform/modules/project
- project.budget in .terraform/modules/project/modules/budget
- project.gsuite_group in .terraform/modules/project/modules/gsuite_group
- project.project-factory in .terraform/modules/project/modules/core_project_factory
- project.project-factory.project_services in .terraform/modules/project/modules/project_services
- project.quotas in .terraform/modules/project/modules/quota_manager
- project.shared_vpc_access in .terraform/modules/project/modules/shared_vpc_access

Initializing the backend...

Initializing provider plugins...
- Finding hashicorp/google-beta versions matching ">= 3.1.0, >= 3.43.0, >= 3.50.0, ~> 4.0, < 4.0.0"...
- Finding hashicorp/null versions matching "~> 2.1"...
- Finding hashicorp/google versions matching ">= 3.43.0, >= 3.50.0, ~> 4.0, < 4.0.0"...
- Finding hashicorp/random versions matching "~> 2.2"...
- Installing hashicorp/null v2.1.2...
- Installed hashicorp/null v2.1.2 (signed by HashiCorp)
- Installing hashicorp/random v2.3.1...
- Installed hashicorp/random v2.3.1 (signed by HashiCorp)
╷
│ Error: Failed to query available provider packages
│ 
│ Could not retrieve the list of available versions for provider
│ hashicorp/google-beta: no available releases match the given constraints >=
│ 3.1.0, >= 3.43.0, >= 3.50.0, ~> 4.0, < 4.0.0
╵

╷
│ Error: Failed to query available provider packages
│ 
│ Could not retrieve the list of available versions for provider
│ hashicorp/google: no available releases match the given constraints >=
│ 3.43.0, >= 3.50.0, ~> 4.0, < 4.0.0
╵
```

Reply:
Hard to work blindly without logs, I cannot fully set up CI on my own with the full GCP organization. I'm afraid to run on a company account :) I have upgraded the versions of the provider, I bet it's failed now with a different miss-configuration since it's a new version.